server/http/interceptors/interceptors.go

```diff
@@ -65,21 +67,34 @@ var contentSecurityPolicyTemplate = strings.Join([]string{
 	"report-uri " + csp.ReportingEndpoint,
 }, ";")
 
-func setContentSecurityPolicy(h http.Header) template.HTMLAttr {
+func setContentSecurityPolicy(h http.Header, host string, useTLS bool) template.HTMLAttr {
 	nonceBytes := make([]byte, 16)
 	_, err := rand.Read(nonceBytes)
 	if err != nil {
 		panic(fmt.Sprintf("Failed to generate nonce: %s", err))
 	}
 	nonce := base64.StdEncoding.EncodeToString(nonceBytes)
+
+	var eventsAPIOrigin string
+	eventsAPIURL := events_api_url.URL()
+	if eventsAPIURL == nil || eventsAPIURL.String() == "" {
+		if useTLS {
+			eventsAPIOrigin = "https://" + host
+		} else {
+			eventsAPIOrigin = "http://" + host
+		}
+	} else {
+		eventsAPIOrigin = eventsAPIURL.Scheme + "//:" + eventsAPIURL.Host
+	}
+
 	// TODO: Enable this by dropping the "-Report-Only" suffix.
-	h.Set("Content-Security-Policy-Report-Only", fmt.Sprintf(contentSecurityPolicyTemplate, nonce))
+	h.Set("Content-Security-Policy-Report-Only", fmt.Sprintf(contentSecurityPolicyTemplate, nonce, eventsAPIOrigin))
```

Comment:
nit: I missed this earlier, but could we build the CSP header using a func that returns a string (like `getContentSecurityPolicy(nonce, eventsAPIOrigin string) string`)  then do string concatenation to fill in the parameterized values? The `%s` approach is a bit difficult to follow because the string is so long, so the reader has to hunt to find the `%s` occurrences and match them up with the parameters here.